### PR TITLE
Failure detector documentation

### PR DESCRIPTION
**Pull request by xnull**
## Overview
https://github.com/CorfuDB/CorfuDB/blob/failure_detector_doc/docs/failure-detector/failure-detector.md

Description:
Failure detector documentation
Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: docs/failure-detector/failure-detector.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Wrong epoch detector
+ - if there is at least one wrong epoch in the cluster, then trigger the correction process. 
+ - check that the slot is unfilled. If the slot is unfilled then fill the slot.
+ - if there was at least one `wrong epoch` response then finish current iteration of a failure detection
+
+### Healing process
+ - Build a ClusterGraph based on the aggregated ClusterState (the cluster state built on the poll report stage)
+ - check if local node is in the unresponsive list.
+ - figure out if we can heal our self:
+   1. the node have to be fully connected which means it have to be connected to all responsive nodes.
```
Comment (hisundar) [inline docs/failure-detector/failure-detector.md]: nit: ...which means it has* to be connected to..

File: docs/failure-detector/failure-detector.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+### Collect poll report
+ - Each node:
+      - pings other nodes 3 times and build local NodeState. 
+      - collects wrong epochs.
+      - collects remote NodeSate's from other nodes.
+ - Aggregate ClusterState from the previous step.     
+
+### Wrong epoch detector
+ - if there is at least one wrong epoch in the cluster, then trigger the correction process. 
+ - check that the slot is unfilled. If the slot is unfilled then fill the slot.
```
Comment (hisundar) [inline docs/failure-detector/failure-detector.md]: Just curious, What does it mean to fill a slot?

Comment (xnull) [reply docs/failure-detector/failure-detector.md]: done

File: docs/failure-detector/failure-detector.md (hunk 1 above)
Comment (xnull) [reply docs/failure-detector/failure-detector.md]: done

File: docs/failure-detector/failure-detector.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+## _Failure detector_
+
+Main component in the failure detector is RemoteMonitoringService.
+The scheduler starts RemoteMonitoringService every 3 seconds (by default).
```
Comment (zalokhan) [inline docs/failure-detector/failure-detector.md]: Every 1 second.
Changed by #1845

Comment (zalokhan) [reply docs/failure-detector/failure-detector.md]: @xnull Please address all comments before merging your PR.

Comment (xnull) [reply docs/failure-detector/failure-detector.md]: maybe I shouldn't even tell a particular number,  it's an implementation specific thing.

**Pull request merged**